pkg/blockbuilder/blockbuilder.go

```diff
@@ -416,46 +384,36 @@ func (b *BlockBuilder) shipperDir() string {
 	return filepath.Join(b.cfg.BlocksStorageConfig.TSDB.Dir, "shipper")
 }
 
-func (b *BlockBuilder) commitOffset(ctx context.Context, part int32, offset int64) (returnErr error) {
-	defer func() {
-		if returnErr != nil {
-			level.Error(b.logger).Log("msg", "failed to commit offset to Kafka", "err", returnErr, "partition", part, "offset", offset)
-			// TODO(codesome): add metric
-		}
-	}()
-
-	toCommit := kadm.Offsets{}
-	toCommit.AddOffset(b.cfg.Kafka.Topic, part, offset, -1)
-
-	committed, err := kadm.NewClient(b.kafkaClient).CommitOffsets(ctx, b.cfg.Kafka.ConsumerGroup, toCommit)
-	if err != nil {
-		return err
-	} else if !committed.Ok() {
-		return committed.Error()
-	}
-
-	committedOffset, _ := committed.Lookup(b.cfg.Kafka.Topic, part)
-	level.Debug(b.logger).Log("msg", "offset successfully committed to Kafka", "offset", committedOffset.At)
-	// TODO(codesome): add this metric
-	// r.lastCommittedOffset.Set(float64(committedOffset.At))
-
-	return nil
+func (b *BlockBuilder) commitOffset(ctx context.Context, checkpointRec *kgo.Record) (err error) {
+	// TODO(codesome): add metric
+	return b.kafkaClient.CommitRecords(ctx, checkpointRec)
```

Comment:
I was looking at adding metadata to the commit right now, and looks like, to do so, we need to use the earlier CommitOffsets() from kadm. Metadata is given part of `kadm.Offset`.